README.md

```diff
@@ -61,8 +61,6 @@
 
```

Comment:
3. :orange_circle: [OPTIONAL]

-` views/balances/index.html.erb`

![optimazition](https://github.com/abeltsew/rich_dad_wallet/assets/26736582/0ffba4ed-5d45-48f5-9d25-151f109a7555)

 The balance.items.map { |item| item.amount }.sum method has to iterate through the entire list of items in the balance, which can be slow if there are a lot of items.

My suggestion is to create a total_amount column in the balances table. This would allow you to quickly calculate the total amount of the balance by simply querying the database.
Here is how you can achieve :arrow_down: 

```
class Item < ApplicationRecord 
 # your existing code here
 after_save :update_balance_total_amount

  private

  def update_balance_total_amount
    balances.each do |balance|
      balance.increment!(:total_amount, amount)
    end
  end
end
```

- Add a new migration
```
class AddTotalAmountToBalances < ActiveRecord::Migration[7.0]
  def change
    add_column :balances, :total_amount, :integer, default: 0
  end
end
```

- In your `views/balances/index.html.erb` line 29 you can just access directly the total_amount by reading the related column.

` <p class="text-bold text-xl">$<%= balance.total_amount %></p>`

The code that I provided in the Item model is a good way to implement this change. The after_save callback will be triggered whenever an item is saved, and the update_balance_total_amount method will be responsible for incrementing the total_amount column in the balance.

This approach is more efficient than the current one because it only has to iterate through the list of items once. This can make a big difference if there are a lot of items in the balance.

I would recommend making this change to your code. It will improve the performance of your application and make it more efficient.